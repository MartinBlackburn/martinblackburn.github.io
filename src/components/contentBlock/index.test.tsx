// libraries
import { render } from "@testing-library/react";

// component under test
import ContentBlock from ".";

describe("ContentBlock component", () => {
    test("renders correctly with image left", () => {
        const TextComponent = () => {
            return <p>some text</p>;
        };

        const { asFragment } = render(
            <ContentBlock title="Climbing" imagePath="climbing.png" imageRight={false}>
                <TextComponent />
            </ContentBlock>
        );

        expect(asFragment()).toMatchSnapshot();
    });

    test("renders correctly with image right", () => {
        const TextComponent = () => {
            return <p>some text</p>;
        };

        const { asFragment } = render(
            <ContentBlock title="Climbing" imagePath="climbing.png" imageRight={true}>
                <TextComponent />
            </ContentBlock>
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
