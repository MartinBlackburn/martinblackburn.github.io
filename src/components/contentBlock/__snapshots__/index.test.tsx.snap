// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentBlock component renders correctly with image left 1`] = `
<DocumentFragment>
  <div
    class="contentBlock"
  >
    <div
      class="contentBlock__image"
    >
      <img
        alt="Climbing"
        src="climbing.png"
      />
    </div>
    <div
      class="contentBlock__content"
    >
      <h2
        class="contentBlock__title"
      >
        Climbing
      </h2>
      <p>
        some text
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ContentBlock component renders correctly with image right 1`] = `
<DocumentFragment>
  <div
    class="contentBlock contentBlock--imageRight"
  >
    <div
      class="contentBlock__image"
    >
      <img
        alt="Climbing"
        src="climbing.png"
      />
    </div>
    <div
      class="contentBlock__content"
    >
      <h2
        class="contentBlock__title"
      >
        Climbing
      </h2>
      <p>
        some text
      </p>
    </div>
  </div>
</DocumentFragment>
`;
